osu.Game/OsuGameBase.cs

```diff
@@ -395,14 +395,14 @@ public async Task Import(params string[] paths)
             }
         }
 
-        public virtual async Task Import(Stream stream, string filename)
+        public virtual async Task Import(params ImportTask[] tasks)
         {
-            var extension = Path.GetExtension(filename)?.ToLowerInvariant();
+            var extension = Path.GetExtension(tasks.First().Path).ToLowerInvariant();
```

Comment:
silently assuming that all imported files will be of the same extension is not safe to do here and will result in import failures if the user doesn't happen to uphold this assumption. each file should be either handled separately, or they should be grouped by extension.

Reply:
That's actually how we do it on `master` at the moment, for best or worst (see method above):

https://github.com/ppy/osu/blob/28674cedd6849ebe4bee8417cad3ed8fe385b2ac/osu.Game/OsuGameBase.cs#L389

Reply:
Guess it's a good opportunity to fix both import paths in this PR

Reply:
You can try, but I'm pretty sure it was done that way with reason. Also if you're changing the other case I'd prefer it in a second PR.

Reply:
From what I can tell this `.First()` call originates from https://github.com/ppy/osu/commit/e0d28564d0d69b4132f6dea94f1e2a162d18c2e4. Before it was checking (arguably more properly) if all extensions were the same before proceeding. Can't see any discussion of that change in the associated PR.